### PR TITLE
NEW JOB: Blueshield Officer (V2, hope this works)

### PR DESCRIPTION
**Pull request by swissloaf**
https://github.com/yogstation13/Yogstation/pull/10042 [Original, closed due to merge conflicts and fixed here]

This adds the Blueshield officer to the game, a consequence of staff discussion over a couple of days

The Blueshield officer is responsible for the protection of all of the Command staff (with Captain being the priority). His office replaces the Command EVA room. Several new sprites have been added and most likely will be updated. You can view them in the sprite check below.

> Most of your time will be spent shadowing those you have been assigned to protect, making sure that they're alive and well, and generally staring at a crew monitor to make sure they're still on sensors. Your job is to keep everyone in Command, as well as CentCom VIPs, alive.
> 
> One very important detail of your job is that you are not Security. Do not let your Security (and Detective) access fool you. Your ID lets you reach every single Command Office on the station, not just the Head of Security. You should not be concerned with the goings-on at Security, nor should you be doing their job for them. You are not a Security Officer with a private office and snazzy outfit. You are a bodyguard.

- [x]  Plasmaman outfit
- [x] Regular outfit
- [x] Port outfits 
- [x] Exempt blueshield from all antags
- [x] Finish office on maps other than default map
- [x] Additional locker or spawn resources
- [x] Solve issue of racism towards nonhumans for this particular role
- [x] figure out why job selection is fucked and wide
- [x] Blueshield Coat issue
- [x] HUDglasses icon 
- [x] Make changes if its human only

image
(character sprite is shitty but just because it is large)

# Changelog

Edit this changelog for changes that are noticeable by the players. Remove it if this isn't the case. If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name. Prefix the PR title with [admin] if it's something admin related. Prefix the PR title with [s] if you are fixing an exploit so it is not announced on discord and the server.

:cl:  
rscadd: Adds a new job: Blueshield Officer 
/:cl:

### CODE REVIEW

**Comment by Darkstick** [comment]
I hope they'll be played as bodyguards, not just sec except blue

**Comment by swissloaf** [comment]
@JamieD1 Please do not let this die again as I have actually been assed to go ahead and fix it

File: code/game/area/Space_Station_13_areas.dm [hunk 1]
```diff
@@ -310,6 +310,10 @@ NOTE: there are two lists of areas in the end of this file: centcom and station
 
 /area/crew_quarters/heads
 	airlock_wires = /datum/wires/airlock/command
+	
+/area/blueshield
+	name = "Blueshield Office"
+	icon_state = "blueshield"
```
Comment (Bibby0110) [inline code/game/area/Space_Station_13_areas.dm:316]:
```suggestion
	icon_state = "blueshield"
	airlock_wires = /datum/wires/airlock/command

```

File: code/modules/jobs/job_types/blueshield.dm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	auto_deadmin_role_flags = DEADMIN_POSITION_HEAD|DEADMIN_POSITION_SECURITY
+	department_head = list("Captain")
+	department_flag = ENGSEC
+	faction = "Station"
+	total_positions = 1
+	spawn_positions = 1
+	supervisors = "the Captain"
+	selection_color = "#ddddff"
+	req_admin_notify = 1
+	space_law_notify = 1
```
Comment (Bibby0110) [inline code/modules/jobs/job_types/blueshield.dm]:
```suggestion
	space_law_notify = TRUE
```

File: code/modules/jobs/job_types/blueshield.dm [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	flag = BLUESHIELD
+	auto_deadmin_role_flags = DEADMIN_POSITION_HEAD|DEADMIN_POSITION_SECURITY
+	department_head = list("Captain")
+	department_flag = ENGSEC
+	faction = "Station"
+	total_positions = 1
+	spawn_positions = 1
+	supervisors = "the Captain"
+	selection_color = "#ddddff"
+	req_admin_notify = 1
```
Comment (Bibby0110) [inline code/modules/jobs/job_types/blueshield.dm]:
```suggestion
	req_admin_notify = TRUE
```

**Review by Partheo — APPROVED** [review]

**Comment by JackKindred** [comment]
this should be good

**Review by patpol4 — APPROVED** [review]
This is not by any means good but it is acceptable after the fixes. Map approved

**Review by JamieD1 — CHANGES_REQUESTED** [review]
I put a hold on the other PR. Making a new one to bypass that hold is not really going to work. I even said I would fix any conflicts that happen due to the hold.

**Comment by swissloaf** [comment]
it's like when I call google customer service and get put on hold for 3 months

File: code/modules/jobs/job_types/blueshield.dm [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	req_admin_notify = TRUE
+	space_law_notify = TRUE
+	exp_requirements = 300
+	exp_type = EXP_TYPE_CREW
+	exp_type_department = EXP_TYPE_SECURITY
+
+	outfit = /datum/outfit/job/blueshield
+
+	access = list(ACCESS_BLUESHIELD, ACCESS_SECURITY, ACCESS_SEC_DOORS, ACCESS_BRIG, ACCESS_COURT, ACCESS_MEDICAL, ACCESS_ENGINE, ACCESS_MAINT_TUNNELS, ACCESS_EXTERNAL_AIRLOCKS, ACCESS_TELEPORTER, ACCESS_HEADS, ACCESS_CAPTAIN, ACCESS_ROBOTICS, ACCESS_CMO, ACCESS_RD, ACCESS_QM, ACCESS_WEAPONS, ACCESS_CE, ACCESS_HOP, ACCESS_HOS, ACCESS_KEYCARD_AUTH, ACCESS_CLONING)
+	minimal_access = list(ACCESS_BLUESHIELD, ACCESS_SECURITY, ACCESS_SEC_DOORS, ACCESS_BRIG	, ACCESS_COURT, ACCESS_MEDICAL, ACCESS_ENGINE, ACCESS_MAINT_TUNNELS, ACCESS_EXTERNAL_AIRLOCKS, ACCESS_TELEPORTER, ACCESS_HEADS, ACCESS_CAPTAIN, ACCESS_ROBOTICS, ACCESS_CMO, ACCESS_RD, ACCESS_QM, ACCESS_WEAPONS, ACCESS_CE, ACCESS_HOP, ACCESS_HOS, ACCESS_KEYCARD_AUTH, ACCESS_CLONING)
```
Comment (SomeguyManperson) [inline code/modules/jobs/job_types/blueshield.dm]: wouldn't keycard auth allow a single head with a blueshield to red alert or do other stuff keycards are supposed to require two heads to agree on for

File: code/modules/jobs/job_types/blueshield.dm [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+/datum/job/blueshield/proc/OmegaStationChanges()
+	return TRUE
+
+/datum/outfit/job/blueshield
+	name = "Blueshield"
+	jobtype = /datum/job/blueshield
+	
+	uniform = /obj/item/clothing/under/rank/blueshield
+	suit = /obj/item/clothing/suit/armor/vest/blueshield
+	gloves = /obj/item/clothing/gloves/combat
```
Comment (SomeguyManperson) [inline code/modules/jobs/job_types/blueshield.dm]: combat gloves are fireproof+shockproof gloves not sure they'd be a good fit for a roundstart role that is fairly close to security in terms of equipment

File: code/modules/jobs/job_types/_job.dm [hunk 6]
```diff
@@ -128,7 +128,7 @@
 	if(!H)
 		return FALSE
 	
-	if(CONFIG_GET(flag/enforce_human_authority) && (title in GLOB.command_positions))
+	if(CONFIG_GET(flag/enforce_human_authority) && (title in GLOB.command_positions) && (title != "Blueshield"))
```
Comment (SomeguyManperson) [inline code/modules/jobs/job_types/_job.dm:131]: has to be a better way of making this work this is kind of snowflake

**Review by SomeguyManperson — COMMENTED** [review]
concerned this is gonna end up as just "security officer with more access" for most people who play it

**Comment by ghost** [comment]
This is a bad idea considering its easy for captains to play like sec, still

**Comment by swissloaf** [comment]
@JamieD1 @SomeguyManperson what is going on with this

**Comment by Tipy1802** [comment]
It’s been like 5 months... How long does it take to write into the rules “Blueshield shouldn’t validhunt unless a head of staff is directly in danger” 
Isn’t that what the HC was waiting on? I mean at first it was the staff RP rules meeting then the townhall then the RP rules and now what are you even waiting for?

File: code/modules/jobs/job_types/blueshield.dm (hunk 4 above)
Comment (swissloaf) [reply code/modules/jobs/job_types/blueshield.dm]: yes I didnt mean to add that woopsies

File: code/modules/jobs/job_types/_job.dm (hunk 6 above)
Comment (swissloaf) [reply code/modules/jobs/job_types/_job.dm:131]: well, this is really the only position that would ever be "command" without human restrictions. if it ever comes that there is another position or if human authority is disabled, we can just remove this

File: code/modules/jobs/job_types/blueshield.dm (hunk 5 above)
Comment (swissloaf) [reply code/modules/jobs/job_types/blueshield.dm]: any recommendations on lesser gloves?

Comment (swissloaf) [reply code/modules/jobs/job_types/blueshield.dm]: changed to black gloves

**Comment by SomeguyManperson** [comment]
@swissloaf fix the merge conflicts and I'll testmerge this for a bit 
personally don't think it's going to work the way it's intended to since giving someone weapons and expecting them to not validhunt doesn't tend to work

**Comment by Yogbot-13** [comment]
#### Test Merge Deployed

##### Server Instance
main

##### Revision
Origin: f1ccd63a36b575f6b21d51678f597a016dea4341
Pull Request: b3b9113c718291b979c200b8e9b69c7bf01ccedf
Server: ff51cbb8edef33f36a4b21e1679264ce72f404e8

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 77b18567de776b91f70821b0cb89efcea50e5638
Pull Request: b3b9113c718291b979c200b8e9b69c7bf01ccedf
Server: 74d901727666ecdb4a83807850b62b7ae4596cc2

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: d9fa53dbf20373162cd911ee6c408579cbb08dd4
Pull Request: b3b9113c718291b979c200b8e9b69c7bf01ccedf
Server: 71059923354c07cd59b391346f037e2cc0a88638

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: d9fa53dbf20373162cd911ee6c408579cbb08dd4
Pull Request: 56e1dbaf4b9b78ea10322e6ad8505d68030f23fc
Server: e241a029bff0a862598d4cf5330270ab7a61901c

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: ef0db3a25ba81dee1c39ea8fc8162a7a4a752ce0
Pull Request: 56e1dbaf4b9b78ea10322e6ad8505d68030f23fc
Server: aaf2a4de3a62a53112d314c3f9281454f06ce656

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 61be296c7775d9ea88e9b8a729456b211e4cc029
Pull Request: 56e1dbaf4b9b78ea10322e6ad8505d68030f23fc
Server: 7556b2e6eebbdd8cd977546ea7280d0bb63bd23f

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 61be296c7775d9ea88e9b8a729456b211e4cc029
Pull Request: 49454b2d5217150b692296c08c8a692e8c5d7875
Server: 30cee7883419fedc2b14d05329f2362084d4a522

**Comment by Mqiib** [comment]
No jumpskirt smh

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 846b6d0591d4fdfa09eb9564a32ffc4691283565
Pull Request: 49454b2d5217150b692296c08c8a692e8c5d7875
Server: a3380909465cd8b2082b69723d93038188c356e5

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 846b6d0591d4fdfa09eb9564a32ffc4691283565
Pull Request: edf75896b82fc47279218b63fdaede073b538506
Server: 952eb07ef4c61276755bc1d018d27c948afb0e36

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 846b6d0591d4fdfa09eb9564a32ffc4691283565
Pull Request: f35b871fb1967fe23290c878ef38914d6a789458
Server: 9a2ec9b6425c6d90690405f9c3b94f4666402c94

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: dd1d7fc33764ab1f7a35cf63a176bb8ea99c5bb7
Pull Request: f35b871fb1967fe23290c878ef38914d6a789458
Server: 14cdaf78532bb345dde5558d0304548c1bc44f79

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: dd1d7fc33764ab1f7a35cf63a176bb8ea99c5bb7
Pull Request: d82045501ab8413ef0fe00d86a03c217fdfb08d1
Server: 38094ac42f2645436ff266e4acf28dbe0a1101bc

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: a6e2415c469ba1af6e29b54aa239ed65b8c36ee7
Pull Request: d82045501ab8413ef0fe00d86a03c217fdfb08d1
Server: d24b215e0e8c6c5423888ae49e4f29a9cf171b3c

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 66b7150aeab900c3f31c14b1bd1e42b0bf923c18
Pull Request: d82045501ab8413ef0fe00d86a03c217fdfb08d1
Server: ef2e93fba57b1a0477b759c6d1a9a04f2aa0c565

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 158c00ddf79f3de3a5da5964b55ac17a89978863
Pull Request: d82045501ab8413ef0fe00d86a03c217fdfb08d1
Server: b8c32c664435221e3bc7144827e3654e90e4e6dc

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: ca59602e0478d0927aad7a9b1fd4711bde1f9c82
Pull Request: d82045501ab8413ef0fe00d86a03c217fdfb08d1
Server: c2377bfb63991a30acd46b8f81123ed6b04133b5

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 39c610f599c05ab55c9551c8328eb9cd5b838117
Pull Request: d82045501ab8413ef0fe00d86a03c217fdfb08d1
Server: 576b980d7860d3a2d8af75956984af14a64580ef

**Comment by Yogbot-13** [comment]
#### Test Merge Updated

##### Server Instance
main

##### Revision
Origin: 39c610f599c05ab55c9551c8328eb9cd5b838117
Pull Request: d82045501ab8413ef0fe00d86a03c217fdfb08d1
Server: ce13dee3d6e20b89dc7daedfb859ee58550b7dcf

**Comment by Jarod1200** [comment]
Delta map lost its hop locker and ian because of the placement of the blueshield office

**Comment by patpol4** [comment]
wew

**Review by SomeguyManperson — CHANGES_REQUESTED** [review]
make sure there is at least one head of staff before blueshields can be selected or rolled since their entire job revolves around protecting heads of staff

**Comment by swissloaf** [comment]
> make sure there is at least one head of staff before blueshields can be selected or rolled since their entire job revolves around protecting heads of staff

I can not code this and Bibby said that this would be very hard to change considering how jobs are selected

also I disagree but what you want

**Comment by Cartlord** [comment]
So if you can't ensure Heads will be present when Blueshields spawn, I'd like to ask: what exactly is one meant to DO, if one is a Blueshield Officer and there are no Heads of Staff present?

**Comment by UselessTheremin** [comment]
swiss is gone, seems like nobody is going to resolve these conflicts, rip blueshield.
tbh i was sold on it until it got testmerged and it just turned into either
a. acting cap every round
b. sec officer with bridge access
c. literally nothing, disappearing 5 minutes into the round

**Comment by alexkar598** [comment]
Note to self: Steal PR and make blueshield a more medical focused role

**Comment by alexkar598** [comment]
Note to others: Stale

**Pull request closed without merging**